### DISCUSSION TITLE
requestLEScan fails on Android 12

### DISCUSSION THREAD

**Discussion by bravesoul349** [capacitor-community/bluetooth-le#437] [Q&A] [answered]
I am not sure if it's a bug or if I have done something wrong so that's why I am starting here.

On initializing the plugin with androidNeverForLocation set to false, the users get two options for location permission on Android 12, either select Precise location or approximate location. On selecting a precise location everything works fine but on selecting the approximate location the scan never completes (possibly an error because the fine location is required if you are not asserting neverForLocation https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#declare-android12-or-higher). 

Is it possible to force ACCESS_FINE_LOCATION on sdk 29 or above
and ACCESS_COARSE_LOCATION on 28 or below

https://developer.android.com/guide/topics/connectivity/bluetooth/permissions#declare-android11-or-lower

**Comment by pwespi** [comment, marked as answer]
Which version of the plugin are you using?

On Android 12 (SDK version 31), the plugin is already asking for `ACCESS_FINE_LOCATION`, see [here](https://github.com/capacitor-community/bluetooth-le/blob/main/android/src/main/java/com/capacitorjs/community/plugins/bluetoothle/BluetoothLe.kt#L87-L111).

However, the user is still presented both options unfortunately, see [here](https://developer.android.com/training/location/permissions#approximate-request). If the user selects "approximate", the `initialize` call should fail and you can react to that. E.g. calling `initialize` again should request an upgrade to precise location.

Hope this helps.
